zebra-rpc/src/methods/tests/snapshot.rs

```diff
@@ -0,0 +1,242 @@
+    let (response, _) = futures::join!(get_raw_transaction, mempool_req);
+    let get_raw_transaction = response.expect("We should have a GetRawTransaction struct");
+
+    snapshot_rpc_getrawtransaction(get_raw_transaction, network);
+
+    // `getaddresstxids`
+    let get_address_tx_ids = rpc
+        .get_address_tx_ids(GetAddressTxIdsRequest {
+            addresses: addresses.clone(),
+            start: 1,
+            end: 10,
+        })
+        .await
+        .expect("We should have an vector of strings");
+    snapshot_rpc_getaddresstxids(get_address_tx_ids, network);
+
+    // `getaddressutxos`
+    let get_address_utxos = rpc
+        .get_address_utxos(AddressStrings { addresses })
+        .await
+        .expect("We should have an vector of strings");
+    snapshot_rpc_getaddressutxos(get_address_utxos, network);
+}
+
+/// Snapshot `getinfo` response, using `cargo insta` and RON serialization.
+fn snapshot_rpc_getinfo(info: GetInfo, network: Network) {
+    let settings = insta::Settings::clone_current();
+    let snap_name = format!("{}_get_info", network_string(network));
+    settings.bind(|| {
+        insta::assert_ron_snapshot!(snap_name, info, {
+            ".subversion" => dynamic_redaction(|value, _path| {
+                // assert that the subversion value is user agent
+                assert_eq!(value.as_str().unwrap(), USER_AGENT);
+                // replace with:
+                "[SubVersion]"
+            }),
+        })
+    });
+}
+
+/// Snapshot `getblockchaininfo` response, using `cargo insta` and RON serialization.
+fn snapshot_rpc_getblockchaininfo(info: GetBlockChainInfo, network: Network) {
+    let settings = insta::Settings::clone_current();
+    let snap_name = format!("{}_get_blockchain_info", network_string(network));
+    settings.bind(|| {
+        insta::assert_ron_snapshot!(snap_name, info, {
+            ".estimatedheight" => dynamic_redaction(|value, _path| {
+                // assert that the value looks like a valid height here
+                assert!(u32::try_from(value.as_u64().unwrap()).unwrap() < Height::MAX_AS_U32);
+                // replace with:
+                "[Height]"
+            }),
+        })
+    });
+}
+
+/// Snapshot `getaddressbalance` response, using `cargo insta` and RON serialization.
+fn snapshot_rpc_getaddressbalance(address_balance: AddressBalance, network: Network) {
+    let settings = insta::Settings::clone_current();
+    let snap_name = format!("{}_get_address_balance", network_string(network));
+    settings.bind(|| insta::assert_ron_snapshot!(snap_name, address_balance));
+}
+
+/// Snapshot `getblock` response, using `cargo insta` and RON serialization.
+fn snapshot_rpc_getblock(block: GetBlock, network: Network) {
+    let settings = insta::Settings::clone_current();
+    let snap_name = format!("{}_get_block", network_string(network));
+    settings.bind(|| insta::assert_ron_snapshot!(snap_name, block));
```

Comment:
Test Code Size: Can you please redact the 1.5 kB of block data, after checking it matches the block data test vector?

Reply:
I could be wrong but it seems not possible to do this because the selector does not support it:

https://insta.rs/docs/redactions/#selectors

`getblock` returns raw data without any key which to make the match against.

Reply:
We can use an `assert!()` to check the data, and then a separate redact to remove it.

Like this snapshot test:
https://github.com/ZcashFoundation/zebra/blob/12e81309410cdf56c0c19f3cc668ef0578c52f2f/zebra-state/src/service/finalized_state/zebra_db/block/tests/snapshot.rs#L495

Except it would be something like:
```rust
assert!(block.contains(BLOCK_MAINNET_1_BYTES.to_hex()));
```

Reply:
Honestly it is not clear to me what you want to do here ...

Reply:
See PR #4393, it was a bit trickier than I thought!

(The JSON path for the root struct is `.`)